snuba/split.py

```diff
@@ -121,22 +125,24 @@ def col_split(dataset, body, *args, **kwargs):
             - Second query selects all fields for only those events.
             - Shrink the date range.
         """
-        minimal_query = {**body, 'selected_columns': MIN_COLS}
-
+        # The query function may mutate the request body during query
+        # evaluation, so we need to copy the body to ensure that the query has
+        # not been modified by the time we're ready to run the full query.
+        minimal_query = copy.deepcopy(body)
+        minimal_query.update({'selected_columns': MIN_COLS})
         result, status = query_func(dataset, minimal_query, *args, **kwargs)
+        del minimal_query
```

Comment:
I was going to collapse this into `{**copy.deepcopy(body), 'selected_columns': MIN_COLS}` and inline it as a local, but since we're already copying it, I figured it's better not to copy it again. The `del` is just here so that it is obvious that `minimal_query` isn't reused later in the function body.